src/lib/Microsoft.Health.Events/EventHubProcessor/IEventProcessorClientFactory.cs

```diff
@@ -11,8 +11,6 @@ namespace Microsoft.Health.Events.EventHubProcessor
 {
     public interface IEventProcessorClientFactory
     {
-        EventProcessorClient CreateProcessorClient(BlobContainerClient blobContainerClient, EventProcessorClientFactoryOptions options, EventProcessorClientOptions eventProcessorClientOptions);
-
-        EventProcessorClient CreateProcessorClient(IAzureCredentialProvider provider, BlobContainerClient blobContainerClient, EventProcessorClientFactoryOptions options, EventProcessorClientOptions eventProcessorClientOptions);
+        EventProcessorClient CreateProcessorClient(BlobContainerClient blobContainerClient, EventProcessorClientFactoryOptions options, EventProcessorClientOptions eventProcessorClientOptions, IAzureCredentialProvider provider = null);
```

Comment:
Should the provider be part of the method signature?  I feel like it can be injected into the ctor for the factory and resolved by DI and if it is needed it will be used.  I.E. The factory needs to be resolve any configuration we should support so it should have provider available.

Does the provider need to vary based on the context?  The concern here is we have an option signature and we allow the options to require the Azure provider but invoke could fail to provide causing an error.

Reply:
I think I did this originally as part of the factory instead of the method, but we changed to the method because it might make sense to be able to generate multiple processors or producers with different configs. Right now there is not a requirement to do so but it could add flexibility in the future.

We were thinking it would be better to have 1 factory that could generate pretty much every kind of client by putting IAzureCredentialProvider in the method signature. If IAzureCredentialProvider is part of the factory instead of the method then we may need to generate a new factory for each custom auth credential provider if there is a requirement to do so in the future.